### PR TITLE
Boardfi -55  extend

### PR DESCRIPTION
**Pull request by Marcin3232**

### CODE REVIEW

File: src/components/ui/switch-icon.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  switch (icon) {
+    case IconType.FLAME:
+      selectedIcon = <FlameIcon className={className} />;
+      break;
+    case IconType.HOURGLASS:
+      selectedIcon = <Hourglass className={className} />;
+      break;
+    default:
+      selectedIcon = <div>No matching icon</div>;
+  }
```
Comment (Saffety2137) [inline src/components/ui/switch-icon.tsx]: Why do we have Icon and SwitchIcon, what is the reasoning behind it?

File: src/pages/HomePage.tsx [hunk 2]
```diff
@@ -1,10 +1,51 @@
...
+              }
+            ></Card>
+          </div>
+        </div>
+        <div className="flex items-center justify-center mb-4">
+          {/* TODO  */}
+          {/* <DataTable
+            columns={generateColumns(mockCryptos[0])}
+            data={mockCryptos}
+          /> */}
```
Comment (Saffety2137) [inline src/pages/HomePage.tsx]: If you add new content on the page, make it work with the current content. Don't comment it out, try to adjust the styles of the whole page.

File: tailwind.config.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ module.exports = {
   prefix: "",
   theme: {
     container: {
+      maxWidth: 'none',
```
Comment (Saffety2137) [inline tailwind.config.js]: I don't know why, but this looks crazy at first glance. If you have a good reasoning of doing this. Please write to me why

File: src/components/CryptocurrienciesRate/CryptocurrenciesRate.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          .slice(0, halfData)
+          .map((cryptoCurrenciesRateView, index) => (
+            <CryptocurrenciesRateRow
+              key={index}
+              cryptoCurrienciesRateView={cryptoCurrenciesRateView}
+              index={index}
+              showCup={true}
+              showCodicon={showCodicon}
+              className="mr-5"
+            />
```
Comment (Saffety2137) [inline src/components/CryptocurrienciesRate/CryptocurrenciesRate.tsx]: I'd rather have this slice logic used inside of the CryptocurrenciesRateRow component.
Add a helper.tsx file and store these data processing operations there.

Also it does not look safe with cutting half data. You are assuming you will get 10 objects from backend which might not be the case. This Component is also non responsive. Add styles so it is at least wrapping nicely. Make it less depeneded on the data delivered and just take first 10 object and display them nicely.

File: src/components/CryptocurrienciesRate/CryptocurrenciesRateRow.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            className="w-6 h-6 rounded-sm overflow-hidden mr-3"
+            alt={`${name} icon`}
+          />
+          {index === 0 && showCup && (
+            <SolarCupBold className="absolute -top-1 right-1 rounded-sm overflow-hidden" />
+          )}
+        </div>
+
+        <span className="mr-2">{name}</span>
+        {showCodicon && codicon && <CodiconFoldUp />}
```
Comment (Saffety2137) [inline src/components/CryptocurrienciesRate/CryptocurrenciesRateRow.tsx]: Try using grid instead maybe, or figure out a way so it behaves better on different resolutions. Icons are not fixed and on a less wide screens they look like shit.

File: src/components/shared/Card.tsx [hunk 6]
```diff
@@ -1,32 +1,45 @@
...
           </h3>
         </div>
       </div>
       <div className="card relative border-transparent border-solid border-4 rounded-xl min-h-64 w-full overflow-visible">
-        <FlameIcon className="absolute w-34 h-49 top-[-25px] left-[-18px] flex-shrink-0 object-cover" />
-        <div className="p-10">
-          <CryptocurrenciesRate></CryptocurrenciesRate>
+        <Icon
+          type={cryptoCurrenciesRateCardView.icon}
+          className="absolute w-34 h-49 top-[-25px] left-[-18px] flex-shrink-0 object-cover"
```
Comment (Saffety2137) [inline src/components/shared/Card.tsx]: Icon is a bit off position. Allign it as it is in Figma.

**Review by Kasperekx — APPROVED** [review]

**Pull request closed without merging**